### ISSUE TITLE
Multi-tab feature

### ISSUE THREAD

**Issue by toludwig** [xournalpp/xournalpp#1971]
It is a little unexpected that on opening/creation of a new file, the one currently worked on is closed.
To be able to edit two files, I need two have instances of Xournal++ open.
This is a quite frequent scenario as I annotate several lectures.

It would be convenient to be able to have multiple tabs open just like in a PDF viewer or in a browser.

The "Recent Documents" list is a less convenient way of switching, but in my case, this does not update at all (see #1970).

Tabs don't necessarily have to appear at the top but could also be included in the sidebar.

**Comment by LittleHuba** [comment]
You should be able to just have two instances of Xournal++ running simultaneously. We are currently unable to provide a tab feature but will do so in the rewrite. But that will take some time until it is up to productive use.

**Comment by idotobi** [comment]
@LittleHuba : I think what @toludwig refers to is that when you use `File -> Open` to open  `2.xopp` from your already running instance with `1.xopp` open, then you just replace `1.xopp` with `2.xopp` in your already running instance.
Thus, you need to start a new xournalpp instance from outside of xournalpp if you want to have both open simultaneously, which actually can be a more elaborate process when having your screen already flipped.
_At least that's what happened to me sometimes in the past. 😅_

Possible Solutions I'd see:
 1. When calling `File -> Open` or `File -> Recent` show a dialog (similar to `This document is not saved yet`) with the Options: a) Open in new Window b) open in current Window.
 2. Change the default to always open in new window
 3. Leave the default

**Comment by LittleHuba** [comment]
I'd see solution 1 as the sole solution here. (2) is infuriating if you have just opened a window of Xournal++ and want to open a document. (3) doesn't fix it if this is what he actually wanted to say (which I'm not sure it is).

**Comment by Febbe** [comment]
Yes a tab feature would be too much work. I think we should implement version one. This is also exactly what clion/jetbrain IDEs do for example.

**Comment by idotobi** [comment]
Agreed.

(2) could be implemented less infuriating by checking first if no changes were made yet to the default page, but that would be probably more confusing than helpful.
(3) Yes that this would mean keeping the status quo. (I just wanted to add it to make it explicit, as of course it's almost always an option).

**Comment by idotobi** [comment]
@toludwig : Would (1) solve your problem, or at least be a good workaround for the time being?

**Comment by toludwig** [comment]
> @toludwig : Would (1) solve your problem, or at least be a good workaround for the time being?

Indeed, that would be a good workaround. Thanks for the suggestions :+1:

**Comment by idotobi** [comment]
Cool, then I'll give it a shot.

**Comment by TsukuyomiDev** [comment]
> We are currently unable to provide a tab feature but will do so in the rewrite. But that will take some time until it is up to productive use.

Is there any Update regarding this rewrite or the multi tab feature?

**Comment by jakelardtrial-source** [comment]
Been over 5 years now, any chance this gets implemented?
